indexing-service/src/main/java/io/druid/indexing/common/task/AbstractTask.java

```diff
@@ -179,4 +183,24 @@ public int hashCode()
   {
     return toolbox.getTaskActionClient().submit(new LockListAction());
   }
+
+  @Override
+  @JsonProperty
+  public Map<String, Object> getContext()
+  {
+    return context;
+  }
+
+  @Override
+  public Object getContextValue(String key)
+  {
+    return context == null ? null : context.get(key);
+  }
+
+  @Override
+  public <ContextType> ContextType getContextValue(String key, ContextType defaultValue)
+  {
+    Object retVal = getContextValue(key);
+    return retVal == null ? defaultValue : (ContextType) retVal;
```

Comment:
Can you add a unit test for this case:

Have an interface which is expected from the context. But specify an implementation of that interface as the default value.

Does the casting and type detection work correctly? More specifically, in such a case is `<ContextType>` determined from the return or the parameter? 

It may need something like `public <ContextType, ContextTypeSuper extends ContextType> ContextType getContextValue(String key, ContextTypeSuper defaultValue)`